filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
```

Comment:
I was thinking of having that be a multi-filter given the initial tag rules use case - you can select tag is A OR B using glob, but you can't do A AND B, since we're strict on tag names. The other use cases can use a multi filter - I'll export the NewMultiFilter function.